misc/metrics/src/identify.rs

```diff
@@ -134,12 +135,12 @@ impl super::Recorder<libp2p_identify::Event> for Metrics {
             }
             libp2p_identify::Event::Received { peer_id, info, .. } => {
                 {
-                    let mut protocols: Vec<String> = info
+                    let mut protocols = info
                         .protocols
                         .iter()
+                        .filter_map(|p| Protocol::try_from_owned(p.to_owned()).ok())
```

Comment:
Long term `libp2p-identify` would expose `Protocol`s instead of `String`s in its identification payload and thus this conversion would no longer be needed, right?